public/index.html

```diff
@@ -0,0 +1,90 @@
+             closeBtn.classList.add('hidden')
+            }
+        socket.on('welcome', msg => {
+            socket.emit('time delay', Date.now())
+            document.querySelector('h1').innerHTML = msg;
+            document.querySelector('button').textContent = notificationCount;
+        });
+        document.querySelector('#submit').addEventListener('click', (e) => {
+           e.preventDefault();
+           const userId = document.querySelector('input').value;
+           socket.emit('join notification', {id: `notification_${userId}`});
+           document.querySelector('form').reset()
+        })
+        socket.on('approved', notification => {
+            socket.emit('time delay', notification.time)
+            notificationCount += 1;
+            document.querySelector('button').textContent = notificationCount;
+            document.querySelector('#alert').innerHTML = `
+           <div class="flex justify-between text-green-200 shadow-inner rounded p-3 bg-green-600">
+                <p class="self-center">
+                <strong>${notification.message}</strong>
+                <a href="${notification.link}" class="hover:underline text-white cursor-pointer">Click here</a>
+                </p>
+                <strong class="text-xl lighn-center cursor-pointer alert-del" onclick="handleClick()">&times;</strong>
+           </div>
+        `
+        })
+        socket.on('rejected', notification => {
+            socket.emit('time delay', notification.time)
+            notificationCount += 1;
+            document.querySelector('button').textContent = notificationCount;
+            document.querySelector('#alert').innerHTML = `
+            <div class="flex justify-between text-green-200 shadow-inner rounded p-3 bg-green-600">
+                <p class="self-center">
+                <strong>${notification.message}</strong>
+                <a href="${notification.link}" class="hover:underline text-white cursor-pointer">Click here</a>
+                </p>
+                <strong class="text-xl lighn-center cursor-pointer alert-del" onclick="handleClick()">&times;</strong>
+           </div>
+           `
+        })
+        socket.on('pending', notification => {
+            socket.emit('time delay', notification.time)
+            notificationCount += 1;
+            document.querySelector('button').textContent = notificationCount;
+            document.querySelector('#alert').innerHTML = `
+            <div class="flex justify-between text-green-200 shadow-inner rounded p-3 bg-green-600">
+                <p class="self-center">
+                <strong>${notification.message}</strong>
+                <a href="${notification.link}" class="hover:underline text-white cursor-pointer">Click here</a>
+                </p>
+                <strong class="text-xl lighn-center cursor-pointer alert-del" onclick="handleClick()">&times;</strong>
+           </div>
+           `
+        })
+    </script>
+</body>
+</html>
```

Comment:
Good for demonstrating, but this will ultimately be a redundant file. You can integrate the notification feature in the frontend and use postman for approval if the approval page is not yet integrated with the back end. Good work!

Reply:
It can't  be such easy right now because we don't have login integrated yet! Therefore, there could be no way to check the logged user to decode id from the token since no login integrated.